float64[] angles
float64[] gradients
float64[] radii
float64 freq
float64 intensity
